pkg/controllermanager/controller/shoot/maintenance/reconciler.go

```diff
@@ -190,6 +200,12 @@ func (r *Reconciler) reconcile(ctx context.Context, log logr.Logger, shoot *gard
 		if err != nil {
 			// continue execution to allow other maintenance activities to continue
 			workerLog.Error(err, "Could not maintain Kubernetes version for worker pool")
+		} else {
+			if maintainedShoot.Spec.Provider.Workers[i].Kubernetes != nil && maintainedShoot.Spec.Provider.Workers[i].Kubernetes.Kubelet != nil {
+				if reasons := maintainFeatureGates(&maintainedShoot.Spec.Provider.Workers[i].Kubernetes.Kubelet.KubernetesConfig, *maintainedShoot.Spec.Provider.Workers[i].Kubernetes.Version, fmt.Sprintf("spec.provider.workers[%d].kubernetes.kubelet.featureGates", i)); len(reasons) > 0 {
+					operations = append(operations, reasons...)
+				}
+			}
```

Comment:
What was the reason to maintain the worker pool specific feature gates here and not in `maintainFeatureGatesForShoot`? If possible I would vote to move this logic in `maintainFeatureGatesForShoot`. Otherwise logic for the same thing is split in different places.

Reply:
The `maintainFeatureGatesForShoot ` calls `maintainFeatureGates` internally for `kubeAPIServer`, `kubeControllerManager`, `kubeScheduler`, `kubeProxy` and `kubelet`. For worker, we only have `kubelet`, so I am calling `maintainFeatureGates` directly.  And at the point where `maintainFeatureGatesForShoot` is called, the worker kubernetes versions are not maintained yet.

Reply:
Can't we call then `maintainFeatureGatesForShoot` after the worker kubernetes versions are maintained?
If possible, I prefer `maintainFeatureGatesForShoot` to maintain all feature gates. Otherwise this handling for kubelet is easy to miss and my feeling is that it does not much feet as abstraction there.